### ISSUE TITLE
Being Kicked Out of a Dungeon ( For no real reason except a bug )

### ISSUE THREAD

**Issue by CaptainJLPv2** [Wynncraft/Issues#8554]
This bug nearly happens in every dungeon ( Probably not Corrupted / Decriept Sewers ).
I know dungeons have a system that is similar to combat-log but it kicks you out of the dungeon so you don't exploit it by unloading the chunks and getting back in, so all the mobs don't respawn for couple seconds, Which I think its fair but not in this scenario due to a bug : 

The bug happens when a structure in a dungeon does not get replaced, such as the decaying bridge in sand swept tombs, which would leave you stuck in there, usually a player would just switch to another world, but the player can't due to the dungeon combat-log system, the player will instead be teleported to the front of the dungeon, which made it a complete waste of time, and effort.

There is one way I found to bypass this bug and that is waiting for a server restart, which no one is going to wait for.

So far I've only seen this bug in these two dungeons

- Fallen Factory ( Middle Tower in Boss Battle Stage will not disappear, but barriers are still at the top preventing the player from insterting the cores to break the monitor. The tower also has no hopper at the bottom ).
- Corrupted / Sand Swept Tombs ( Decaying Bridge will not respawn )

This also can happen when you defeat a boss and you don't get your rewards, thus you are stuck in the boss room. Usually the boss would respawn, but very rarely it will not, As shown in #8296.

This bug has been in wynncraft for a very long time and it is time it gets fixed.

**Comment by colin350** [comment]
These bugs are related to build/cmds, and will not be patched through until the next map update. Asking them to be fixed now is complicated especially if we are working on a new update, as Grian wrote in a post on the forums. Pushing the map now will leak parts of the map which we wish not to be. 

Switch servers by itself during a boss fight is an exploit, and that probably won't be changed. 

However, I have gone in and added a failsafe for the decaying bridge in both SST and CSST. However, I'm rather confident that those changes were not patched in since I did them a while after Corkus was released. 

I have yet to look into the Fallen Factory one, but it's on our long todo list.

**Comment by CaptainJLPv2** [comment]
I have found screenshot evidence of the bug in fallen factory.
I cant really make a screenshot of the barriers at the top as their invisible.
![2017-07-11_17 41 14](https://user-images.githubusercontent.com/28000844/28056636-e119cdb4-6660-11e7-8b2d-2905a9a71085.png)
No hopper at the bottom.
![2017-07-11_17 37 03](https://user-images.githubusercontent.com/28000844/28056651-f085bbc8-6660-11e7-98c1-e3fb8dfcfe47.png)
The tower being there before I collected a single thing.

**Comment by CaptainJLPv2** [comment]
I think the cause of this bug in fallen factory is, killing the boss before the tower even despawns, so the chunks become unloaded, therefore whatever is supposed to remove the tower is unloaded meaning it probably can't run how its supposed to.

**Comment by LordHelmchen98** [comment]
You are right this happens if the boss in Fallen Factory dies very fast. There are some redstone mechanics that break if the chunks get unloaded before the process is complete.
However Fallen Factory is the only dungeon that I know of with a structure like that in the boss part. You should not get kicked out of the dungeon if you switch servers at the decaying bridge.
